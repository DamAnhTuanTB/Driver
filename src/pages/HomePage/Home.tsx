// import { useFetchApi } from "hooks/useFetchApi";
// import { useQuery } from "react-query";
// import { homeApi } from "./services";

import HomeCenter from "components/HomeCenter";
import { withRouter } from "react-router-dom";

function Home() {

    // eslint-disable-next-line react-hooks/rules-of-hooks
    // const { data } = useQuery('notification-detail', () => useFetchApi(homeApi.getHome()), {
    // });

    return (
        <div className="flex min-h-screen items-center justify-center">
            <HomeCenter />
        </div>
    );
}

export default withRouter(Home);
