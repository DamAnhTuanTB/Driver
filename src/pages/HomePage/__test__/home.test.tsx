import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import Home from '../Home';

describe('Test able to render', () => {
    test('able to render Home', async () => {
        setUp();
        expect(screen.queryByTestId('welcome-test')).not.toBeNull();
    });
    test('snap home', () => {
        const { asFragment } = setUp();

        expect(asFragment()).toMatchSnapshot();
    });

});

const setUp = () => {
    return render(
        <QueryClientProvider client={new QueryClient()}>
            <Home />
        </QueryClientProvider>
    );
};
