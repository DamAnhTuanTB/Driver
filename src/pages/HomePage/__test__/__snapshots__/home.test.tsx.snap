// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test able to render snap home 1`] = `
<DocumentFragment>
  <div
    data-testid="welcome-test"
  >
    welcome to driver team
  </div>
</DocumentFragment>
`;
